src/language-html/print-preprocess.js

```diff
@@ -408,4 +410,18 @@ function addIsSpaceSensitive(ast, options) {
   });
 }
 
+function markTsScript(ast, options) {
+  if (options.parser === "vue") {
+    ast.walk((node) => {
```

Comment:
I think this is really bad, we are walking big vue SFC just to find the root script.(I think `.walk()` can't quit).
If we want allow ts only if ts script found, I think we can found them from the root.

Also, I'm not sure if we really need this check, I like things be safe, but I think we can always pase it as ts? I don't use ts myself, I'm fine with both add or remove this check.

Reply:
Thank you.

It is unsafe to use TS parser always. For example, The below code is valid as JS and TS.

```ts
const foo = doSomething<T1 | T2>(param)
```

```js
// as JS
const foo = (doSomething < T1) | (T2 > param);
```

TS and JS parse differently for such codes. Therefore, it is safe to parse as TypeScript only when the script tag explicitly specifies the use of TS.

Reply:
I've fixed to avoid using `ast.walk`. [35b2600](https://github.com/prettier/prettier/pull/12584/commits/35b2600c0d3b7d295dfc31ce0877c030bf8af51b)

Reply:
Does it make sense in any case to use `jsx` and `template` together?
IMO you are using `jsx` with the `render`-function and then you do not use the `__vue_expression` and `__vue_event_binding`, or you use `template` and therefore not `jsx`/`tsx`.